web/source/osk/preProcessor.ts

```diff
@@ -30,50 +9,39 @@ namespace com.keyman.osk {
      * 
      * @param       {Object}      e      element touched (or clicked)
      */
-    static clickKey(e: osk.KeyElement, touch?: Touch, layerId?: string, keyDistribution?: text.KeyDistribution) {
+    static clickKey(e: osk.KeyElement, touch?: Touch) {
+      let keyman = com.keyman.singleton;
+      let Lkc = keyman['osk'].vkbd.initKeyEvent(e, touch);
+      if(!Lkc) {
+        return true;
+      }
+
+      return this.raiseKeyEvent(Lkc);
+    }
+
+    static raiseKeyEvent(keyEvent: text.KeyEvent) {
       let keyman = com.keyman.singleton;
       var Lelem = keyman.domManager.getLastActiveElement();
 
       if(Lelem != null) {
         // Handle any DOM state management related to click inputs.
         let outputTarget = dom.Utils.getOutputTarget(Lelem);
         keyman.domManager.initActiveElement(Lelem);
-  
-        // Turn off key highlighting (or preview)
-        keyman['osk'].vkbd.highlightKey(e,false);
```

Comment:
Why is this no longer needed?

Reply:
Elsewhere in this PR: https://github.com/keymanapp/keyman/blob/28fe887f84c2936708f846c7ae88fb5d856ba879/web/source/osk/visualKeyboard.ts#L1543-L1545

`initKeyEvent` always occurs for any OSK-based key event and will run first.  As a result, this deletion removes a redundancy.